### PR TITLE
Integrate BasLogger into Ingester and remove deprecated code

### PR DESCRIPTION
**Pull request by cammv21**
## Description

This PR refactors the WarehouseIngester to clean up and integrate our standard BasLogger, as outlined in issue #252.

Key changes:

- Removed all deprecated require statements and service definitions related to old Notion implementations.
- Replaced all puts error messages with structured BAS_LOGGER.error calls, which now include the error class and backtrace.
- Added BAS_LOGGER.info logging for successful ingestions to track the entity_type and processed item count.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Summary by CodeRabbit

* **Changes**
  * Centralized service registry for many new entity types and removed hardcoded mappings.
  * Ingestion now validates readiness early and skips unprocessable responses.
  * Improved logging and observability: structured success/failure logs and processed counts reported.
  * More robust upsert and error handling so failed records are reported and do not inflate processed counts.
* **Tests**
  * Updated to cover OKR-oriented flows, logging assertions, and new edge-case behaviors.

### CODE REVIEW

File: src/implementations/warehouse_ingester.rb [hunk 1]
```diff
@@ -138,6 +131,45 @@ def upsert(item)
...
+        invoker: 'WarehouseIngester',
+        message: message,
+        context: { action: 'ingest', entity_type: @type }
+      }
+
+      payload[:context][:processed] = processed unless processed.nil?
+
+      payload.merge!(format_error_payload(error, message)) if error
+
+      BAS_LOGGER.send(level, payload)
```
Comment (villalbaluis) [inline src/implementations/warehouse_ingester.rb]: Earlier this morning we was talking about how this is working with the method `send` to clarify this, the doc:
- https://ruby-doc.org/core-3.1.2/Object.html#method-i-send

`Send` is part of the object properties, so it takes the given param and look for the corresponding method.

File: src/implementations/warehouse_ingester.rb [hunk 2]
```diff
@@ -138,6 +131,45 @@ def upsert(item)
       else
         @service.insert(item)
       end
+
+      true
+    end
+
+    def log_ingestion(level, message, processed: nil, error: nil)
```
[the remark this answers is not in the document]
Comment (cammv21) [reply src/implementations/warehouse_ingester.rb]: Thanks, Lucho. That makes sense, good catch.

File: src/implementations/warehouse_ingester.rb [hunk 3]
```diff
@@ -138,6 +131,45 @@ def upsert(item)
...
+      }
+
+      payload[:context][:processed] = processed unless processed.nil?
+
+      payload.merge!(format_error_payload(error, message)) if error
+
+      BAS_LOGGER.send(level, payload)
+    end
+
+    def setup_ingestion_valid
```
[the remark this answers is not in the document]
Comment (cammv21) [reply src/implementations/warehouse_ingester.rb]: Okay, I'll take your recommendation.
